airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx

```diff
@@ -101,22 +116,54 @@ export const ConnectorBuilderStateProvider: React.FC<React.PropsWithChildren<unk
   const lastValidJsonManifest = useMemo(
     () =>
       editorView !== "ui"
-        ? undefined
+        ? jsonManifest
         : builderFormValues === lastValidBuilderFormValues
         ? jsonManifest
         : convertToManifest(lastValidBuilderFormValues),
     [builderFormValues, editorView, jsonManifest, lastValidBuilderFormValues]
   );
 
+  const [testStreamIndex, setTestStreamIndex] = useState(0);
+
+  const [selectedView, setSelectedView] = useState<BuilderView>("global");
+
+  const ctx = {
+    builderFormValues,
+    jsonManifest: derivedJsonManifest,
+    lastValidJsonManifest,
+    yamlManifest,
+    yamlEditorIsMounted,
+    yamlIsValid,
+    testStreamIndex,
```

Comment:
What is the reasoning behind putting `testStreamIndex` in FormState instead of TestState?

Reply:
The test stream index is set in a bunch of places (from the side bar, from the stream config view, from the builder errors hook) - those places do not subscribe to the test state right now. If they would, an update to the test state would re-render most of the UI again.

Reply:
It's a little annoying so many places need to subscribe to the test state, solely because they want to set the test stream index. But actually I found a better way to handle that - instead of explicitly setting testing stream index and view index to the same value, I added a hook to the new test state service to handle that. Makes things much simpler and is a better separation of concerns.